File: Java/RedBlackTrees/com/tree/Node.java
package com.tree;

public class Node{
    public int data;
    public Node left,right,parent;
    public int size,sum;
    public String color;
    public Node(int data){
        this.data=data;
        this.left=null;
        this.right=null;
        this.parent=null;
        this.size=0;
        this.sum=0;
        this.color="B";
    }
}

File: Ruby/RedBlackTrees/node.rb
class Node
  attr_accessor :key, :satellite_data, :p, :left, :right
  def initialize(key, satellite_data, p, left, right)
    @key, @color, @p, @left, @right = key, satellite_data, p, left, right
  end

  #   Node.new(10, nil, nil, nil)
  #   # => #<Node:0x007ffeab2187a0 @key=10, @p=nil, @left=nil, @right=nil>
  def initialize(key, color, p, left, right)
    @key, @color, @p, @left, @right = key, color, p, left, right
  end
end
